src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py

```diff
@@ -1358,9 +1358,89 @@ def load_trend_output_description(
 
     def generate_trend_description(
         document: DescriptionDocument,
-    ) -> case_description.TrendOutputDescription:
+    ) -> case_description.PositionalTrendDescription:
         case_values = to_case_values(document, alfacase_to_case_description)
-        item_description = case_description.TrendOutputDescription(**case_values)
+        item_description = case_description.PositionalTrendDescription(**case_values)
+        return update_multi_input_flags(document, item_description)
+
+    return [
+        generate_trend_description(alfacase_document) for alfacase_document in document
+    ]
+
+
+def load_equipment_trend_description(
+    document: DescriptionDocument,
+) -> List[case_description.EquipmentTrendDescription]:
+    alfacase_to_case_description = {
+        "curve_names": load_value,
+        "element_name": load_value,
+    }
+
+    def generate_trend_description(
```

Comment:
You can extract this internal function used in all load_* functions:

```python
    def generate_trend_description(
        document: DescriptionDocument,
        alfacase_to_case_description,  # Add type
        description_class: Type,
    ) -> case_description.PositionalTrendDescription:
        case_values = to_case_values(document, alfacase_to_case_description)
        item_description = description_class(**case_values)
        return update_multi_input_flags(document, item_description)
```

Reply:
Well noted, just changed it, thanks!